### ISSUE TITLE
TinyMCE textcolor_map

### ISSUE THREAD

**Issue by sshelby1** [WordPress/gutenberg#11147]
**Describe the bug**
When filtering the TinyMCE textcolor_map option the colors are replaced with blanks.

**To Reproduce**
Filter the editor as so:

add_filter('tiny_mce_before_init', 'change_tinymce_color_options');
function change_tinymce_color_options($options) {
	$custom_colours =  '
	"000000", "Black",
    "993300", "Burnt orange",
    "333300", "Dark olive",
    "003300", "Dark green",
    "003366", "Dark azure",
    "000080", "Navy Blue",
    "333399", "Indigo",
    "333333", "Very dark gray",
    "800000", "Maroon",
    $options['textcolor_map'] = '['.$custom_colours.']';
	
    return $options;
}

**Expected behavior**
the custom colors should be visible in the tinymce editor when using a classic editor block

**Screenshots**
![image](https://user-images.githubusercontent.com/44512617/47603284-b8ef6100-d9af-11e8-8ace-cf53360b1819.png)

**Comment by becomevocal** [comment]
On beta version 4.9.9-alpha-43656 on Mac, Chrome latest (70.0.3538.77), with Gutenberg v4.1.1, I can reproduce.

**Steps to reproduce**

Add filter:

```function change_tinymce_colors( $init ) {
    $default_colours = '
        "FF0000", "Red",
        "FF9900", "Amber",
        "33CCCC", "Turquoise",
        "3366FF", "Royal blue",
        "800080", "Purple",
        "FF00FF", "Magenta",
        "FFCC00", "Gold",
        "FFFF00", "Yellow"
        ';
    $custom_colours = '
        "edf1ff", "Color 1",
        "ebebec", "Color 2",
        "f1f8fe", "Color 3",
        "eee6f9", "Color 4"
        ';
    $init['textcolor_map'] = '['.$default_colours.','.$custom_colours.']';
    $init['textcolor_rows'] = 1;
    return $init;
}
add_filter('tiny_mce_before_init', 'change_tinymce_colors');
```

**Screenshot**

<img width="658" alt="screen shot 2018-10-28 at 4 30 58 pm" src="https://user-images.githubusercontent.com/2677921/47621513-568d8200-dacf-11e8-80e4-08b0362ff070.png">

**Notes**

You can see that the colors aren't respected but the number of rows config is.

**Comment by becomevocal** [comment]
@danielbachhuber yep. #10968 fixes the default colors list, however it doesn't show the custom colors properly.

<img width="656" alt="screen shot 2018-10-30 at 8 34 38 am" src="https://user-images.githubusercontent.com/2677921/47731972-7d100200-dc22-11e8-9540-e4f78fccf84a.png">

**Comment by sshelby1** [comment]
I am not sure what change I need to make to test it. It is still broken on my implementation.
    On Tuesday, October 30, 2018, 11:10:06 AM CDT, Nate Stewart <notifications@github.com> wrote:  
 
 
@danielbachhuber yep. #10968 fixes the default colors list, however it doesn't show the custom colors properly.

**Comment by azaozz** [comment]
@becomevocal sorry but I'm not sure what is the intent of the [example code](#issuecomment-433739094). Having `$init['textcolor_rows'] = 1;` means "hide all colors table rows except the first one", and that is what happens. If you remove that setting, the extra colors show up.

**Comment by becomevocal** [comment]
@azaozz you're right :) Seems like this is being fixed in #10968 so this ticket can probably be marked duplicate?

@sshelby1 I got this to work by making the changes to my install as listed here: https://github.com/WordPress/gutenberg/pull/10968/files

**Comment by mtias** [comment]
For reference, this can also be made to work on new blocks like this:
![color](https://user-images.githubusercontent.com/548849/48324131-58e2f680-e60e-11e8-8d32-354f4e776ebc.gif)

**Comment by youknowriad** [comment]
closed by #10968
